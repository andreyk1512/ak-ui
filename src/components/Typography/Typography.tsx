import { TypographyProps } from './Typography.types';

export const Typography = ({ children }: TypographyProps) => <article>{children}</article>;
