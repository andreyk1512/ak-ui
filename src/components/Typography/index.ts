export * from './Typography';
export * from './Typography.types';

import { Text, Title, Paragraph } from './components';
import { Typography as OriginTypography } from './Typography';

export type TypographyProps = typeof OriginTypography & {
  Text: typeof Text;
  Title: typeof Title;
  Paragraph: typeof Paragraph;
};

export const Typography = OriginTypography as TypographyProps;
Typography.Text = Text;
Typography.Title = Title;
Typography.Paragraph = Paragraph;
