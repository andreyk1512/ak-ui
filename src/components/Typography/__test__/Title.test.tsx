import { render } from '@testing-library/react';
import { Typography } from '../Typography';

const checkTitleElement = (container: HTMLElement, htmlElementSelector: string, children: string) => {
  expect(container.querySelector(htmlElementSelector)?.textContent).toBe(children);
  expect(container.firstChild).toMatchSnapshot();
};

describe('Typography.Title', () => {
  it('should correctly render h1 element by default with content', () => {
    const children = 'Lorem ipsum dolor sit amet...';
    const { container } = render(<Typography.Title>{children}</Typography.Title>);

    checkTitleElement(container, 'h1', children);
  });

  it('should correctly render h2 element by default with content', () => {
    const children = 'Lorem ipsum dolor sit amet...';
    const { container } = render(<Typography.Title type={2}>{children}</Typography.Title>);

    checkTitleElement(container, 'h2', children);
  });

  it('should correctly render h3 element by default with content', () => {
    const children = 'Lorem ipsum dolor sit amet...';
    const { container } = render(<Typography.Title type={3}>{children}</Typography.Title>);

    checkTitleElement(container, 'h3', children);
  });

  it('should correctly render h4 element by default with content', () => {
    const children = 'Lorem ipsum dolor sit amet...';
    const { container } = render(<Typography.Title type={4}>{children}</Typography.Title>);

    checkTitleElement(container, 'h4', children);
  });

  it('should correctly render h5 element by default with content', () => {
    const children = 'Lorem ipsum dolor sit amet...';
    const { container } = render(<Typography.Title type={5}>{children}</Typography.Title>);

    checkTitleElement(container, 'h5', children);
  });
});
