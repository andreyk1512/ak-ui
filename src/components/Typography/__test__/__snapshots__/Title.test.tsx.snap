// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography.Title should render the title content as h1 for type 1 by default 1`] = `
<h1
  class="css-1llirg8"
>
  This is a title
</h1>
`;

exports[`Typography.Title should render the title content as h2 for type 2 1`] = `
<h2
  class="css-1skqbxe"
>
  This is a title
</h2>
`;

exports[`Typography.Title should render the title content as h3 for type 3 1`] = `
<h3
  class="css-wcntdc"
>
  This is a title
</h3>
`;

exports[`Typography.Title should render the title content as h4 for type 4 1`] = `
<h4
  class="css-18ii8n4"
>
  This is a title
</h4>
`;

exports[`Typography.Title should render the title content as h5 for type 5 1`] = `
<h5
  class="css-fz6ed2"
>
  This is a title
</h5>
`;
